lib/Sema/CSSyntacticElement.cpp

```diff
@@ -358,16 +358,24 @@ class SyntacticElementConstraintGenerator
   friend StmtVisitor<SyntacticElementConstraintGenerator, void>;
 
   ConstraintSystem &cs;
-  AnyFunctionRef context;
+  DeclContext *dc;
```

Comment:
This is actually moving in opposite direction (it used to be DeclContext originally) but we’d like to support `TapExpr` as a context as well - https://github.com/apple/swift/pull/61746

Reply:
Makes sense, shall I pull in SyntacticElementContext minus `TapExpr` for this PR?

Reply:
Sure, it should be cherry-pickable.

Reply:
How does this look?